actions/actions.go

```diff
@@ -203,10 +203,148 @@ func TeardownSuite(
+func sendFunds(chainlinkNodes []client.Chainlink, network client.BlockchainClient) (map[int]string, error) {
+	chainlinkTransactionAddresses := make(map[int]string)
+	sendFundsErrGroup := new(errgroup.Group)
+	for ni, n := range chainlinkNodes {
+		nodeIndex := ni // https://golang.org/doc/faq#closures_and_goroutines
+		node := n
+		// Send async request to each chainlink node to send a transaction back to the network default wallet
+		sendFundsErrGroup.Go(
+			func() error {
+				primaryEthKeyData, err := node.ReadPrimaryETHKey()
+				if err != nil {
+					return err
+				}
+				nodeBalanceString := primaryEthKeyData.Attributes.ETHBalance
+				if nodeBalanceString != "0" { // If key has a non-zero balance, attempt to transfer it back
+					gasCost, err := network.EstimateTransactionGasCost()
+					if err != nil {
+						return err
+					}
+
+					// TODO: Imperfect gas calculation buffer of 50 Gwei. Seems to be the result of differences in chainlink
+					// gas handling. Working with core team on a better solution
+					gasCost = gasCost.Add(gasCost, big.NewInt(50000000000))
+					nodeBalance, _ := big.NewInt(0).SetString(nodeBalanceString, 10)
+					transferAmount := nodeBalance.Sub(nodeBalance, gasCost)
+					_, err = node.SendNativeToken(transferAmount, primaryEthKeyData.Attributes.Address, network.GetDefaultWallet().Address())
+					if err != nil {
+						return err
+					}
+					// Add the address to our map to check for later (hashes aren't returned, sadly)
+					chainlinkTransactionAddresses[nodeIndex] = strings.ToLower(primaryEthKeyData.Attributes.Address)
+				}
+				return nil
+			},
+		)
+
+	}
+	return chainlinkTransactionAddresses, sendFundsErrGroup.Wait()
+}
+
+// checks that the funds made it from the chainlink node to the network address
+// this turns out to be tricky to do, given how chainlink handles pending transactions, thus the complexity
+func checkFunds(chainlinkNodes []client.Chainlink, sentFromAddressesMap map[int]string, toAddress string) error {
+	err := retry.Do( // Might take some time for txs to confirm, check up on the nodes a few times
+		func() error {
+			log.Debug().Msg("Attempting to confirm chainlink nodes transferred back funds")
+			transactionErrGroup := new(errgroup.Group)
+			for nodeIndex, n := range chainlinkNodes {
+				node := n // https://golang.org/doc/faq#closures_and_goroutines
+				sentFromAddress, nodeHasFunds := sentFromAddressesMap[nodeIndex]
+				// Async check on all the nodes if their transactions are confirmed
+				if nodeHasFunds { // Only if the node had funds to begin with
+					transactionErrGroup.Go(func() error {
+						return confirmTransaction(node, sentFromAddress, toAddress, transactionErrGroup)
+					})
+				} else {
+					log.Debug().Int("Node Number", nodeIndex).Msg("Chainlink node had no funds to return")
+				}
+			}
+
+			return transactionErrGroup.Wait()
+		},
+		retry.Delay(time.Second*5),
```

Comment:
We have two retry libraries, again, one from `ginkgo` and that one

Reply:
I'm assuming you're referring to the Ginkgo `Eventually`? We can switch to that, but it seemed out of place here, as this shouldn't be asserting anything or failing any tests.